elements/lisk-transactions/src/utils/get_block_bytes.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import {
+	BIG_ENDIAN,
+	hexToBuffer,
+	intToBuffer,
+	LITTLE_ENDIAN,
+} from '@liskhq/lisk-cryptography';
+
+import { BlockHeader } from '../transaction_types';
+
+const SIZE_INT32 = 4;
+const SIZE_INT64 = 8;
+
+export const getBlockBytes = (block: BlockHeader) => {
+	const blockVersionBuffer = intToBuffer(
+		block.version,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const timestampBuffer = intToBuffer(
+		block.timestamp,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const previousBlockBuffer = block.previousBlockId
+		? intToBuffer(block.previousBlockId, SIZE_INT64, BIG_ENDIAN)
+		: Buffer.alloc(SIZE_INT64);
+
+	const seedRevealBuffer = Buffer.from(block.seedReveal, 'hex');
+
+	const heightBuffer = intToBuffer(block.height, SIZE_INT32, LITTLE_ENDIAN);
+
+	const maxHeightPreviouslyForgedBuffer = intToBuffer(
+		block.maxHeightPreviouslyForged,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const maxHeightPrevotedBuffer = intToBuffer(
+		block.maxHeightPrevoted,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const numTransactionsBuffer = intToBuffer(
+		block.numberOfTransactions,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const totalAmountBuffer = intToBuffer(
+		block.totalAmount.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const totalFeeBuffer = intToBuffer(
+		block.totalFee.toString(),
```

Comment:
here if and `totalAmount` and `reward` is calling `toString()` but if we create `BlockHeaderJSON`, we dont need toString() anymore

Reply:
I agree but maybe calling `toString()` on a string doesn't hurt much? the implementation will the similar, easy to modify and copy between two places in the future?